File: Questão-7/Escola.py
class Escola:
  def __init__(self, nome):
    self.professores = []
    self.nome = nome

  def add_professor(self, professor):
    self.professores.append(professor)

  def Listar_professores(self):
    print(f"\nLista de Professores {self.nome}:")
    i = 1
    for professor in self.professores:
      print(f"{i} - {professor.nome} Matéria: {professor.materia}")
      i += 1

class Professor:
  def __init__(self, nome, materia):
    self.nome = nome
    self.materia = materia

escola1 = Escola("Escola Boa")
escola2 = Escola("Escola Bacana")

professor1 = Professor("Leonardo", "Matemática")
professor2 = Professor("Hugo", "Ciência")
professor3 = Professor("Fernando", "Química")

escola1.add_professor(professor1)
escola1.add_professor(professor2)
escola1.add_professor(professor3)
escola2.add_professor(professor3)
escola2.add_professor(professor2)
escola2.add_professor(professor1)

escola1.Listar_professores()
escola2.Listar_professores()

File: Questão-7/Escola.go
package main

import (
	"fmt"
)

type Professor struct{
		nome string
		materia string
}

type Escola struct{
		nome string
		professores []Professor
}


func (e *Escola) addProfessor(p Professor){
		e.professores = append(e.professores, p)
}
func (e Escola) ListarProfessores(){
		var i int = 1
		fmt.Printf("\nLista de Professores %s: \n", e.nome)		
		for _, p := range e.professores{
				fmt.Printf("%d - %s Matéria: %s\n", i, p.nome, p.materia)
				i+=1
		}
}



func main() {


	escola1 := Escola{nome:"Escola Boa"}
	escola2 := Escola{nome:"Escola Bacana"}

	professor1 := Professor{nome: "Leonardo", materia: "Matemática"}
	professor2 := Professor{nome: "Hugo", materia: "Ciência"}
	professor3 := Professor{nome: "Fernando", materia: "Química"}

	escola1.addProfessor(professor1)
	escola1.addProfessor(professor2)
	escola1.addProfessor(professor3)

	escola2.addProfessor(professor3)
	escola2.addProfessor(professor2)
	escola2.addProfessor(professor1)

	escola1.ListarProfessores()
	escola2.ListarProfessores()
	
}
